### PR TITLE
Dev to master

### PR DESCRIPTION
**Pull request by NULLx76**

### CODE REVIEW

File: .idea/codeStyles/codeStyleConfig.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<component name="ProjectCodeStyleConfiguration">
+  <state>
+    <option name="PREFERRED_PROJECT_CODE_STYLE" value="Default" />
```
Comment (NULLx76) [inline .idea/codeStyles/codeStyleConfig.xml:3]: Should we keep the `.idea` folder?

File: core/config/config.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+type Builder struct {
+	Config
+}
+
+func (b *Builder) SetDefault() BuilderProcess {
+	b.Config = Config{
+		WebAddr:  		"0.0.0.0:8042",
+		BasePath: 		"/",
+		PubKeyPath: 	"./id_25519.pub",
+		SecretKeyPath:  "./id_25519",
```
Comment (NULLx76) [inline core/config/config.go:47]: indenting seems non consistent, we probably need to run some linting.

File: core/jwt/ecc/keygenutil_test.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package ecc
+
+import (
+	"testing"
+)
+
+func TestKeyGenerationUtil(t *testing.T) {
+	KeyGenerationUtil("stdout", "", "")
```
Comment (NULLx76) [inline core/jwt/ecc/keygenutil_test.go:8]: This test doesn't seem that useful

File: core/passwords/passwords.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	"github.com/trustelem/zxcvbn"
+)
+
+var commonPasswords = []string{
+	"aurum",
+	"finitum",
+}
+
+func VerifyPassword(password string, userinput []string) bool {
+	if len(password) < 8 {
```
Comment (NULLx76) [inline core/passwords/passwords.go:13]: magic number, should be const

File: core/web/AuthEndpoints.go [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+		log.Infof("A user with username \"%s\" has signed up. This is the first user and will get admin privileges.", u.Username)
+		u.Role = db.AdminRoleID
+	}
+
+	// Actually add the user to the db
+	err = e.Repos.CreateUser(u)
+	if err != nil {
+		// look if the error was caused by the username already existing
+		// TODO: This error is SQL specific so should not be handled here probably
+		if strings.HasPrefix(err.Error(), "UNIQUE constraint failed:") {
```
Comment (NULLx76) [inline core/web/AuthEndpoints.go:69]: This is quite ugly and should be changed.

File: docs/apidoc/utils/handlebars_helper.js [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+    Handlebars.registerHelper('startTimer', function(text) {
+        timer = new Date();
+        return '';
+    });
+
+    Handlebars.registerHelper('stopTimer', function(text) {
+        console.log(new Date() - timer);
+        return '';
+    });
+
```
Comment (NULLx76) [inline docs/apidoc/utils/handlebars_helper.js:37]: Do we want to commit generated code?

File: .idea/codeStyles/codeStyleConfig.xml (hunk 1 above)
Comment (jonay2000) [reply .idea/codeStyles/codeStyleConfig.xml:3]: Everything except runconfigs should be removed

File: core/config/config.go (hunk 2 above)
Comment (jonay2000) [reply core/config/config.go:47]: just run formatting

File: core/config/keys.go [hunk 7]
```diff
@@ -0,0 +1,74 @@
+package config
+
+import (
+	"aurum/jwt/ecc"
+	log "github.com/sirupsen/logrus"
+	"os"
+)
+
+// TODO: Testing
```
[the remark this answers is not in the document]
Comment (jonay2000) [reply core/config/keys.go:9]: after, don't have time now

**Pull request merged**